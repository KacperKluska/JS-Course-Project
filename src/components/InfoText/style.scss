section {
  background-color: #999;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: 200px;
  padding: 10px 50px;
  border-bottom: #555 solid 1px;

  header {
    border: none;
    margin: 20px 0px;
  }

  h1 {
    color: black;
    font-size: 2rem;
    background-color: #999;
  }

  #text {
    color: black;
    font-size: 1.2rem;
    padding: 20px 50px;
  }
}
