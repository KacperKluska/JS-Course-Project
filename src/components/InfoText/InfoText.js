import "./style.scss";

export default function InfoText({ title, text }) {
  return (
    <section>
      <header>
        <h1>{title}</h1>
      </header>
      <div id="text">{text}</div>
    </section>
  );
}
