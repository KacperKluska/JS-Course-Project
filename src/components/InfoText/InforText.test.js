import { shallow, render } from "enzyme";
import renderer from "react-test-renderer";
import InfoText from "./InfoText";

let wrapper;

beforeAll(() => {
  wrapper = shallow(
    <InfoText title="Test title">Some test text to be printed</InfoText>
  );
});

test("type test", () => {
  expect(wrapper.type()).toBe("section");
});

test("title tests", () => {
  expect(wrapper.find("header").html()).toContain("<h1>");
  expect(wrapper.find("header").text()).toBe("Test title");
});

test("text test", () => {
  expect(wrapper.find("div").text()).toBe("Some test text to be printed");
});

test("InfoText snapshot", () => {
  const component = renderer.create(
    <InfoText title="Some title">Some placeholder text given</InfoText>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
