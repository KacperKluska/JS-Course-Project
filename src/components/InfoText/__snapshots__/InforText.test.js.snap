// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoText snapshot 1`] = `
<section
  className="infoText"
  id="Some title"
>
  <header>
    <h1>
      Some title
    </h1>
  </header>
  <div
    className="text"
  >
    Some placeholder text given
  </div>
</section>
`;
